pkg/cloud/baremetal/actuators/machine/actuator.go

```diff
@@ -886,6 +898,95 @@ func (a *Actuator) requestPowerOn(ctx context.Context, baremetalhost *bmh.BareMe
 	return err
 }
 
+// isActionTimedOut checks if time since timestamp stored in specified machine annotation did not exceed given timeout
+func isActionTimedOut(machine *machinev1beta1.Machine, annotation string, timeout time.Duration) bool {
+	annotations := machine.Annotations
+	if annotations == nil {
+		return false
+	}
+	startTimeString, exist := annotations[annotation]
+	if !exist {
+		log.Printf("Annotation %q not found on machine %q", annotation, machine.Name)
+		return false
+	}
+	startTime, err := time.Parse(annotationTimestampFormat, startTimeString)
+	if err != nil {
+		log.Printf("Unable to parse time from annotation %q on machine %q", annotation, machine.Name)
```

Comment:
Same question about how this message should be interpreted by a user reading the logs.

I think in this case we're assuming that means the action has not timed out. Why assume that, instead of assuming that it has timed out?

Reply:
This is the same case as one above. The goal of the assumption is to fallback to the old behaviour before adding this feature.

Reply:
OK, the same comment about adding more detail applies.